doc/man3/SSL_CTX_set_session_ticket_cb.pod

```diff
@@ -0,0 +1,141 @@
+=pod
+
+=head1 NAME
+
+SSL_CTX_set_session_ticket_cb,
+SSL_SESSION_get0_ticket_appdata,
+SSL_SESSION_set1_ticket_appdata - manage session ticket application data
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ void SSL_CTX_set_session_ticket_cb(SSL_CTX *ctx,
+                                    tls_generate_session_ticket_cb_fn gen_cb,
+                                    tls_decrypt_session_ticket_cb_fn dec_cb,
+                                    void *arg);
+
+ int SSL_SESSION_set1_ticket_appdata(SSL_SESSION *ss, const void *data, size_t len);
+ void SSL_SESSION_get0_ticket_appdata(SSL_SESSION *ss, void **data, size_t *len);
+
+=head1 DESCRIPTION
+
+SSL_CTX_set_set_session_ticket_cb() sets the application callbacks B<gen_cb>
+and B<dec_cb> that are used by a server to set and get application data stored
+with a session, and placed into a session ticket. The value of B<arg> is passed
+to the callbacks.
```

Comment:
Is it valid for one of them to be NULL?

Reply:
Yes, but it makes little sense.

Reply:
Do you want a check for both NULL/non-NULL?

Reply:
Well, I'm thinking you may not need to know about it at the time the session is decrypted, but at some later time. In which case you might want to supply a NULL dec_cb. You might want to set them both to NULL if they were previously set and you want to unset them. At a bit more of a stretch I suppose there might be scenarios where you don't want to set any app data on future tickets, but want to retrieve it from old ones in which case you might want a NULL gen_cb but set dec_cb.

Reply:
That's kinda what I was thinking, so no change.

Reply:
Should it be mentioned in the documentation that NULL is a permissible value?